src/utils/ucc_coll_utils.c

```diff
@@ -94,6 +94,68 @@ ucc_coll_args_is_mem_symmetric(const ucc_coll_args_t *args,
     return 0;
 }
 
+
+/* If this is the root and the src/dst buffers are asymmetric, the dst needs
```

Comment:
I'm not sure this what we agreed on, if I remember correctly new memory type should match memory type of non root rank buffer. On task completion copy back is not always needed, sometimes we need to copy src buffer but keep original dst buffer